### ISSUE TITLE
Support Targeting Go Versions For Staticcheck

### ISSUE THREAD

**Issue by mneverov** [golangci/golangci-lint#1943]
**Is your feature request related to a problem? Please describe.**
I'd like golangci-lint to support [targeting go versions for staticcheck](https://staticcheck.io/docs/#targeting-go-versions). 
From [go 1.15 release notes](https://golang.org/doc/go1.15#testing):
 `A TestMain function is no longer required to call os.Exit. If a TestMain function returns, the test binary will call os.Exit with the value returned by m.Run.`

```
golangci-lint --version
golangci-lint has version 1.39.0 built from 9aea4aee on 2021-03-26T08:02:53Z
```

When staticcheck run as a separate binary with `-go 1.15` there is no errors, when run via golangci-lint:
`SA3000: TestMain should call os.Exit to set exit code (staticcheck)` is reported.

**Describe the solution you'd like**
An option to configure go version for staticcheck.
